api/src/com/cloud/network/vpc/QuaggaZoneConfig.java

```diff
@@ -0,0 +1,331 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package com.cloud.network.vpc;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.commons.lang.StringUtils;
+
+import com.cloud.exception.InvalidParameterValueException;
+import com.cloud.utils.net.NetUtils;
+
+public class QuaggaZoneConfig {
+
+    private long zoneId;
+    private String protocol;
+    private String ospfArea;
+    private String helloInterval;
+    private String deadInterval;
+    private String retransmitInterval;
+    private String transitDelay;
+    private String authentication;
+    private String password;
+    private String[] ospfSuperCIDRList;
+    private String enabled;
+
+    public static final String s_quaggaProtocol = "quaggaprotocol";
+    public static final String s_ospfArea = "ospfarea";
+    public static final String s_quaggaHelloInterval = "quaggahellointerval";
+    public static final String s_quaggaDeadInterval = "quaggadeadinterval";
+    public static final String s_quaggaRetransmitInterval = "quaggaretransmitinterval";
+    public static final String s_quaggaTransitDelay = "quaggatransitdelay";
+    public static final String s_quaggaAuthentication = "quaggaauthentication";
+    public static final String s_ospfSuperCIDR = "ospfsupercidr";
+    public static final String s_quaggaPassword = "quaggapassword";
+    public static final String s_quaggaEnabled = "quaggaenabled";
```

Comment:
Why not collapse these constants into the ``Params`` enumeration with a lookup method to return the enumeration value for the parameter key?